### PR TITLE
CI: use latest distribution of elastic-package 

### PR DESCRIPTION
**Pull request by mtojek**
Related: https://github.com/elastic/apm-server/issues/8491

Changes: 
* use the latest version of elastic-package to the latest available now
* adjust `genpackage` to set the correct version in `changelog.yml`
* add missing `README.md`
* rename in various places `build/integrations` to `build/packages`

### CODE REVIEW

File: .ci/packaging.groovy [hunk 1]
```diff
@@ -125,21 +125,22 @@ pipeline {
...
         stage('apmpackage') {
           options { skipDefaultCheckout() }
-          when {
-            allOf {
-              // The apmpackage stage gets triggered as described in https://github.com/elastic/apm-server/issues/6970
-              changeset pattern: '(cmd/version.go|apmpackage/.*)', comparator: 'REGEXP'
-              not { changeRequest() }
-            }
-          }
+          // when {
```
Comment (mtojek) [inline .ci/packaging.groovy:128]: Disabling it temporarily for development purposes

File: go.mk [hunk 2]
```diff
@@ -39,8 +45,12 @@ $(STATICCHECK): $(GITROOT)/tools/go.mod
 $(GOLICENSER): $(GITROOT)/tools/go.mod
 	$(GO) build -o $@ -modfile=$< github.com/elastic/go-licenser
 
-$(ELASTICPACKAGE): $(GITROOT)/tools/go.mod
-	$(GO) build -o $@ -modfile=$< -ldflags '-X github.com/elastic/elastic-package/internal/version.CommitHash=anything' github.com/elastic/elastic-package
+# FIXME elastic-package requires Go 1.18 runtime to build, but apm-server uses on 1.17.
```
Comment (jsoriano) [inline go.mk:48]: Is this the reason to stop using the Go toolchain to install elastic-package?

File: internal/otel_collector/service/defaultcomponents/defaults.go [hunk 3]
```diff
@@ -1,16 +1,19 @@
...
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
```
Comment (jsoriano) [inline internal/otel_collector/service/defaultcomponents/defaults.go:16]: Are these license changes expected?

**Review by jsoriano — APPROVED** [review]

File: go.mk (hunk 2 above)
Comment (mtojek) [reply go.mk:48]: Yes, elastic-package doesn't build due to the 1.18 requirement of a kubernetes-sigs library.

File: internal/otel_collector/service/defaultcomponents/defaults.go (hunk 3 above)
Comment (mtojek) [reply internal/otel_collector/service/defaultcomponents/defaults.go:16]: I don't know, let's wait for other reviewers. CI complained about those.

File: go.mk (hunk 2 above)
Comment (axw) [reply go.mk:48]: Is the [`.go-version`](https://github.com/elastic/elastic-package/blob/main/.go-version) file in elastic-package still used? It says 1.17.6. Also, go.mod says `go 1.17`. Also also, I built the repo with Go 1.17 :smile:

Perhaps when you used `go get github.com/elastic/elastic-package@latest`, you were using a Go 1.18 toolchain and it pulled a too-new version of the k8s dependencies?

File: internal/otel_collector/service/defaultcomponents/defaults.go (hunk 3 above)
Comment (axw) [reply internal/otel_collector/service/defaultcomponents/defaults.go:16]: No they should not. Can you please point out the CI failures?

Comment (axw) [reply internal/otel_collector/service/defaultcomponents/defaults.go:16]: Never mind, I can reproduce locally. Probably related to the bump in go-licenser. I've got a new PR open, I'll look into it there.

**Comment by axw** [comment]
I've created a separate PR for updating elastic-package: https://github.com/elastic/apm-server/pull/8712

Perhaps close this one, or otherwise repurpose it for adding a timestamp to the package.

**Comment by mtojek** [comment]
Closing in favor of https://github.com/elastic/apm-server/pull/8712.

**Pull request closed without merging**